src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,117 @@
+
+        Response Code should be 200
+        """
+        # register devices
+        device_a = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_a.json')))
+        device_c = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_c.json')))
+        # Register the devices and assert the Response
+        cbsd_ids = self.assertRegistered([device_a, device_c])
+
+        # Create grant requests
+        grant_0 = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_1 = json.load(
+            open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_0['cbsdId'] = cbsd_ids[0]
+        grant_1['cbsdId'] = cbsd_ids[1]
+        # Request for non-overlapping frequency spectrum
+        grant_0['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3620000000.0,
+            'highFrequency': 3630000000.0
+        }
+        grant_1['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3640000000.0,
+            'highFrequency': 3650000000.0
+        }
+        request = {'grantRequest': [grant_0, grant_1]}
+        # Send grant requests
+        response = self._sas.Grant(request)['grantResponse']
+        # Check registration response
+        grant_ids = []
+        for resp in response:
+            self.assertValidResponseFormatForApprovedGrant(resp)
+            self.assertEqual(resp['response']['responseCode'], 0)
+            grant_ids.append(resp['grantId'])
+        del request, response
+
+        # Trigger the SAS to generate the activity dump
+        self._sas_admin.TriggerFullActivityDump()
+
+        # Get dump 
+        response = self._sas.GetFullActivityDump()
+        # Verify the response with  FullActivityDump.schema.json Object schema
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        generation_date_time = datetime.strptime(response['generationDateTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+        if generation_date_time >= (datetime.utcnow() - timedelta(minutes=1)):
+            for activity_dump_file in response['files']:
+                # Verify the record type 
+                if activity_dump_file['recordType'] == 'cbsd':
+                    # Verify the response files with ActivityDumpFile.schema.json Object schema
+                    self.assertContainsRequiredFields("ActivityDumpFile.schema.json", activity_dump_file)
+                    # Verify the url format is valid (pip install validators)
+                    self.assertTrue(validators.url(activity_dump_file['url']))
+                    # Verify the link to download the file dump is valid
+                    self.assertEqual(urlopen(activity_dump_file['url']).getcode(), 200)
+
+                    # Get data in file
+                    data = urlopen(activity_dump_file['url']).read()
```

Comment:
How about using PyCurl itself?
`
def _DownloadFile(url, ssl_cert, ssl_key):
  with io.BytesIO() as file_:
    conn = pycurl.Curl()
    conn.setopt(conn.VERBOSE, 3)
    conn.setopt(conn.SSLVERSION, conn.SSLVERSION_TLSv1_2)
    conn.setopt(conn.SSLCERTTYPE, 'PEM')
    conn.setopt(conn.SSLCERT, ssl_cert)
    conn.setopt(conn.SSLKEY, ssl_key)
    conn.setopt(conn.CAINFO, CA_CERT)
    conn.setopt(conn.SSL_CIPHER_LIST, ':'.join(CIPHERS))
    conn.setopt(pycurl.FOLLOWLOCATION, 1)
    conn.setopt(pycurl.MAXREDIRS, 5)
    conn.setopt(pycurl.CONNECTTIMEOUT, 60)
    conn.setopt(pycurl.TIMEOUT, 300)
    conn.setopt(pycurl.NOSIGNAL, 1)
    conn.setopt(pycurl.URL, url)
    conn.setopt(pycurl.WRITEDATA, file_)
    conn.perform()
    conn.close()
    return file_.getvalue()
`
May be addition of this function in sas.py?